tox.ini

```diff
@@ -126,8 +126,9 @@ commands_pre =
     pip install -e astropy[test]
     pip install -r {env_tmp_dir}/requirements.txt
     pip freeze
-commands =
-    pytest astropy/astropy/io/misc/asdf --open-files --run-slow --remote-data
+commands=
+    pytest astropy/astropy/io/misc/asdf --open-files --run-slow --remote-data \
+        -W "ignore::asdf.exceptions.AsdfDeprecationWarning:asdf.extension._legacy"
```

Comment:
What part of astropy is emitting this warning?

Reply:
<details>
<summary>

```
FAILED astropy/astropy/io/misc/asdf/tags/fits/tests/test_fits.py::test_backwards_compat - asdf.exceptions.AsdfDeprecationWarning: default_extensions is deprecated. Please see the new extension API https://asdf.readthedocs.io/en/stable/asdf/extending/converters.html
FAILED astropy/astropy/io/misc/asdf/tags/table/tests/test_table.py::test_backwards_compat - asdf.exceptions.AsdfDeprecationWarning: default_extensions is deprecated. Please see the new extension API https://asdf.readthedocs.io/en/stable/asdf/extending/converters.html
```

</summary>

```
_______________________________________________________________________________________________________________ test_backwards_compat _______________________________________________________________________________________________________________

    @pytest.mark.filterwarnings(
        "ignore:The property AsdfFile.blocks has been deprecated:asdf.exceptions.AsdfDeprecationWarning"
    )
    def test_backwards_compat():
        """
        Make sure that we can continue to read FITS HDUs that use the schema from
        the ASDF Standard.

        This test uses the examples in the fits schema from the ASDF Standard,
        since these make no reference to Astropy's own fits definition.
        """

        def check(asdffile):
            assert isinstance(asdffile["example"], fits.HDUList)

>       run_schema_example_test("stsci.edu", "asdf", "fits/fits", "1.0.0", check)

astropy/astropy/io/misc/asdf/tags/fits/tests/test_fits.py:59:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
astropy/astropy/io/misc/asdf/tags/tests/helpers.py:12: in run_schema_example_test
    uri = asdf.extension.default_extensions.extension_list.tag_mapping(tag)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

name = 'default_extensions'

    def __getattr__(name):
        if name == "default_extensions":
>           warnings.warn(
                "default_extensions is deprecated. "
                "Please see the new extension API "
                "https://asdf.readthedocs.io/en/stable/asdf/extending/converters.html",
                AsdfDeprecationWarning,
            )
E           asdf.exceptions.AsdfDeprecationWarning: default_extensions is deprecated. Please see the new extension API https://asdf.readthedocs.io/en/stable/asdf/extending/converters.html

../../../asdf/extension/__init__.py:82: AsdfDeprecationWarning
_______________________________________________________________________________________________________________ test_backwards_compat _______________________________________________________________________________________________________________

    @pytest.mark.filterwarnings(
        "ignore:The property AsdfFile.blocks has been deprecated:asdf.exceptions.AsdfDeprecationWarning"
    )
    def test_backwards_compat():
        """
        Make sure that we can continue to read tables that use the schema from
        the ASDF Standard.

        This test uses the examples in the table schema from the ASDF Standard,
        since these make no reference to Astropy's own table definition.
        """

        def check(asdffile):
            assert isinstance(asdffile["example"], table.Table)

>       run_schema_example_test("stsci.edu", "asdf", "core/table", "1.0.0", check)

astropy/astropy/io/misc/asdf/tags/table/tests/test_table.py:259:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
astropy/astropy/io/misc/asdf/tags/tests/helpers.py:12: in run_schema_example_test
    uri = asdf.extension.default_extensions.extension_list.tag_mapping(tag)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

name = 'default_extensions'

    def __getattr__(name):
        if name == "default_extensions":
>           warnings.warn(
                "default_extensions is deprecated. "
                "Please see the new extension API "
                "https://asdf.readthedocs.io/en/stable/asdf/extending/converters.html",
                AsdfDeprecationWarning,
            )
E           asdf.exceptions.AsdfDeprecationWarning: default_extensions is deprecated. Please see the new extension API https://asdf.readthedocs.io/en/stable/asdf/extending/converters.html

../../../asdf/extension/__init__.py:82: AsdfDeprecationWarning
============================================================================================================== short test summary info ==============================================================================================================
FAILED astropy/astropy/io/misc/asdf/tags/fits/tests/test_fits.py::test_backwards_compat - asdf.exceptions.AsdfDeprecationWarning: default_extensions is deprecated. Please see the new extension API https://asdf.readthedocs.io/en/stable/asdf/extending/converters.html
FAILED astropy/astropy/io/misc/asdf/tags/table/tests/test_table.py::test_backwards_compat - asdf.exceptions.AsdfDeprecationWarning: default_extensions is deprecated. Please see the new extension API https://asdf.readthedocs.io/en/stable/asdf/extending/converters.html
```

</details>